csharp/src/Google.Protobuf/Collections/MapField.cs

```diff
@@ -666,6 +690,45 @@ public void MergeFrom(CodedInputStream input)
                     }
                 }
 
+                [SecuritySafeCritical]
+                public void InternalMergeFrom(ref ParseContext ctx)

```

Comment:
The `public void MergeFrom(CodedInputStream input)` on MessageAdapter has no test coverage.

Reply:
refactored a bit to deduplicate the code and added a test.